src/gaxios.ts

```diff
@@ -59,15 +59,20 @@ export class Gaxios {
   async request<T = any>(opts: GaxiosOptions = {}): GaxiosPromise<T> {
     opts = this.validateOpts(opts);
     try {
-      const res = await fetch(opts.url!, opts);
-      const data = await this.getResponseData(opts, res);
-      const translatedResponse = this.translateResponse(opts, res, data);
-      if (!opts.validateStatus!(res.status)) {
+      let translatedResponse: GaxiosResponse<T>;
+      if (opts.adapter) {
+        translatedResponse = await opts.adapter<T>(opts);
+      } else {
+        const res = await fetch(opts.url!, opts);
+        const data = await this.getResponseData(opts, res);
+        translatedResponse = this.translateResponse<T>(opts, res, data);
```

Comment:
I'm curious, should `gaxios` still translate the response 
 and response data if there's an adapter? It looks like [axios does](https://github.com/axios/axios/blob/master/lib/core/dispatchRequest.js#L63).

Reply:
agreed

I have a usecase where I simply want to wrap the default adapter rather than replacing it completely (for my usecase, I want to rate limit requests)